fix: segmentation fault, non initialized pointers in dictziplib.hpp

mandriota opened the pull request:
In method void DictData::close() both if (this->chunks) and if (this->offsets) checks can pass even if no allocated memory is assigned to pointers, because both pointers were not initialized to anything, it is UB, so free were called on uninitialized values. Initializing both values to nullptr guarantees that both checks will fail if pointers were not modified, and free is not called.

mandriota committed:
fix: segmentation fault, non initialized pointers in dictziplib.hpp

In method void DictData::close() both if (this->chunks) and if (this->offsets) checks can pass even if no allocated memory is assigned to pointers, because both pointers were not initialized to anything, it is UB, so free were called on uninitialized values. Initializing both values to nullptr guarantees that both checks will fail if pointers were not modified, and free is not called.

Dushistov commented on src/dictziplib.hpp:
Because of sdcv uses c++11 `chunks` and `ofsets` can be initialized right here.
So if in the future, if somebody add one more constructor, but will not appear again.

mandriota replied:
Are you speaking about member initializer list? I am currently reading from Github's mobile client and don't see if there is any attached code: later will check on PC.

mandriota replied:
Probably I understood what you meant: you are speaking about that it is possible to assign default value to attribute at the same line with its declaration.

mandriota replied:
@Dushistov Updated.

mandriota committed:
fix: segmentation fault, non initialized pointers in dictziplib.hpp

In method void DictData::close() both if (this->chunks) and if (this->offsets) checks can pass even if no allocated memory is assigned to pointers, because both pointers were not initialized to anything, it is UB, so free were called on uninitialized values. Initializing both values to nullptr guarantees that both checks will fail if pointers were not modified, and free is not called.

Dushistov commented:
Thanks for PR.
